compiler/src/iree/compiler/Dialect/Flow/Transforms/OutlineDispatchRegions.cpp

```diff
@@ -65,12 +65,42 @@ static std::string getLinalgOpLoopRanges(linalg::LinalgOp op) {
   return outputString;
 }
 
+static std::string operandTypeToString(OpOperand *opOperand) {
+  auto operandType = opOperand->get().getType();
+  std::string outputString;
+  llvm::raw_string_ostream sstream(outputString);
+  if (auto shapedType = dyn_cast<ShapedType>(operandType)) {
+    shapedType.getElementType().print(sstream);
+  } else {
+    operandType.print(sstream);
+  }
+  return outputString;
+}
+
+// Returns a string like "f32xf32" representing a linalg op's types for each
+// operands.
+static std::string getLinalgDataTypes(linalg::LinalgOp op) {
+  std::string outputString;
+  llvm::raw_string_ostream sstream(outputString);
+  llvm::interleave(
+      op.getDpsInputOperands(),
+      [&](OpOperand *opOperand) { sstream << operandTypeToString(opOperand); },
+      [&] { sstream << "x"; });
+  sstream << "x";
+  llvm::interleave(
+      op.getDpsInitOperands(),
+      [&](OpOperand *opOperand) { sstream << operandTypeToString(opOperand); },
+      [&] { sstream << "x"; });
```

Comment:
Input and init operands seem like they should be separated with a different symbol than "x". Are both useful enough to include?

Reply:
They're definitely both useful to include. I think it would be confusing to show one and not the other (some ops can have more outputs than inputs, like `linalg.generic`.

Reply:
Matching the MLIR print and/or our flags would be least ambiguous.
`ins(tensor<32xf32>, tensor<1x160x160x32xf32>) outs(tensor<1x160x160x32xf32>)`
`input=2x2xi32 input=2x2xi32 output=2xi32`

Both of those styles end up being pretty verbose though.

> Do you have a suggestion for different symbol? 

"in" and "out" maybe?

---

What's the most critical information to include in a summary?

In my view, we're summarizing an entire dispatch, which may include multiple linalg ops. The critical pieces of information to me would be
* What sorts of ops are included in this dispatch (e.g. named linalg ops like matmul)
* How heavy is the computation in this dispatch? (how many reads/writes/flops/etc.)

If a program has mixed types (some float, some quantized, some double precision), that could also be useful to know, but for different reasons.

I'm trying to decide if it makes sense to extend the single op summary (like here, adding more type information), or if summary characters would be better utilized by pulling in information from the other ops in the dispatch (like total estimated flops as a sum of all the loop ranges)

Reply:
I'd vote no for the verbose stuff. Ultimately a quick summary of the type info is whats critical here. Most dispatches have a single type, but many do mix types (see example). 

As it is now, the shape and defining op are already ideal. The only other information of interest is the op type, which can have material effects on the dispatch from the consideration of performance and (at least on CUDA) pipeline selection. 

I don't think its necessary to increase the summary for multiple ops. Unless you have a specific use case for this, I don't see it and not really the focus here. 

For the separator between ins/outs, I was expecting a single char? (keep it short). Else in the case when all types match, its just one type.

Reply:
I'd at least add a comment explaining that "input" and "init" types are included together. 

Or maybe write "to" between input and output types?
```
forward_dispatch_1_generic_1x128x768_i32xi32xf32        <-- what you have now
forward_dispatch_1_generic_1x128x768_i32xi32_to_f32     <-- looks alright
forward_dispatch_1_generic_1x128x768_f32_from_i32xi32   <-- more characters
forward_dispatch_1_generic_1x128x768_in_i32xi32_out_f32 <--- naaah
```